src/App.tsx

```diff
@@ -1,26 +1,53 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { getPreparedTodos } from './utils/todos';
+import { useAppSelector } from './app/hooks';
+import { useDispatch } from 'react-redux';
+import { actions } from './features/todos';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const inputQuery = useAppSelector(state => state.filter.query);
+  const selectQuery = useAppSelector(state => state.filter.status);
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useDispatch();
+
+  const preparedTodos = getPreparedTodos(todos, { inputQuery, selectQuery });
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => {
+      dispatch(actions.setTodos(todosFromServer));
+      setIsLoading(false);
+    });
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {!isLoading ? <TodoList todos={preparedTodos} /> : <Loader />}
```

Comment:
Change it for better readability(Up to u)
```suggestion
              {
                !isLoading 
                  ? <TodoList todos={preparedTodos} />
                  : <Loader />
              }
```